src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertConcatenationToInterpolatedStringRefactoringProvider.cs

```diff
@@ -157,7 +157,23 @@ private Task<Document> UpdateDocumentAsync(Document document, SyntaxNode root, S
                 }
                 else
                 {
-                    content.Add(generator.Interpolation(piece.WithoutTrivia()));
+                    var firstInterpolationToken = piece.GetFirstToken();
+                    var lastInterpolationToken = piece.GetLastToken();
+
+                    if (firstInterpolationToken.RawKind == startToken.RawKind && lastInterpolationToken.RawKind == endToken.RawKind)
+                    {
+                        // If the piece like $"foo {a} goo: }",
```

Comment:
your example is slightly confusing.  was there a reason to ahve that trailing `}`?

Reply:
No it is a mistake
I have remove it